README-PRE-5.md

```diff
@@ -0,0 +1,374 @@
+# kubernetes-client
+
+[![Build Status][build]](https://travis-ci.org/godaddy/kubernetes-client) [![Greenkeeper badge][greenkeeper]](https://greenkeeper.io/)
+
+[greenkeeper]: https://badges.greenkeeper.io/godaddy/kubernetes-client.svg
+[build]: https://travis-ci.org/godaddy/kubernetes-client.svg?branch=master
+
+Simplified [Kubernetes API](http://kubernetes.io/) client for Node.js.
+
+This kubernetes-client API is deprecated. See [README.md](README.md)
+for the new API that has better support for all Kubernetes API
+endpoints.
+
+## Installation
+
+Install via npm:
+
+```console
+$ npm i kubernetes-client --save
+```
+
+## Examples
+
+kubernetes-client provides access to all the Kubernetes objects and
+includes some niceties for writing simpler code.
+
+### Basics
+
+kubernetes-client maps the URI paths in the Kubernetes API to
+sequences of objects chained together via properties and ending in a
+function. For example, to GET the `ReplicationController` named
+'http-rc' in the `Namespace` 'my-project':
+
+```js
+const Api = require('kubernetes-client');
+const core = new Api.Core({
+  url: 'http://my-k8s-api-server.com',
+  version: 'v1'  // Defaults to 'v1'
+});
+
+function print(err, result) {
+  console.log(JSON.stringify(err || result, null, 2));
+}
+
+core.namespaces('default').replicationcontrollers('http-rc').get(print);
+```
+
+kubernetes-client supports the Extensions API group. For example, GET
+the `Deployment` named `http-deployment`:
+
+```js
+const ext = new Api.Extensions({
+  url: 'http://my-k8s-api-server.com',
+  version: 'v1beta1'  // Defaults to 'v1beta1'
+});
+
+ext.namespaces('default').deployments('http-deployment').get(print);
+```
+
+kubernetes-client provides a helper to get in-cluster config and accessing the API from a Pod:
+
+```js
+const Api = require('kubernetes-client');
+const core = new Api.Core(Api.config.getInCluster());
+```
+
+and a helper to get the current-context config from `~/.kube/config`:
+
+```js
+const Api = require('kubernetes-client');
+const core = new Api.Core(Api.config.fromKubeconfig());
+```
+
+### **Experimental** support for promises and async/await
+
+kubernetes-client has **experimental** support for promises. If you
+omit callbacks an HTTP method function (*e.g.*, `.get`), it will
+return a promise.
+
+```js
+core.namespaces('default').replicationcontrollers('http-rc').get()
+  .then(result => print(null, result));
+```
+
+or with `async/await`:
+
+```js
+print(null, await core.namespaces('default').replicationcontrollers('http-rc').get());
```

Comment:
This calls attention to `print` being a little weird - maybe just console.log the result?